libraries/lib-components/EffectInterface.h

```diff
@@ -131,10 +140,73 @@ class COMPONENTS_API EffectDefinitionInterface  /* not final */ : public Compone
 
    //! Report names of factory presets
    virtual RegistryPaths GetFactoryPresets() = 0;
+
    //! Change settings to the preset whose name is `GetFactoryPresets()[id]`
    virtual bool LoadFactoryPreset(int id) = 0;
    //! Change settings back to "factory default"
    virtual bool LoadFactoryDefaults() = 0;
+
+   //! @}
+
+   /*! @name settings
+    Interface for saving and loading externalized settings.
+    All methods are const!
+    */
+   //! @{
+   //! Produce an object holding new, independent settings
+   virtual Settings MakeSettings() const = 0;
+
+   //! Store settings as keys and values
+   /*!
+    @return true on success
+    */
+   virtual bool SaveSettings(
+      const Settings &settings, CommandParameters & parms) const = 0;
+
+   //! Restore settings from keys and values
+   /*!
+    @return true on success
+    */
+   virtual bool LoadSettings(
+      CommandParameters & parms, Settings &settings) const = 0;
```

Comment:
This is meant to copy/map a CommandParameters to a Settings (at least that's how I understand it)
Should it not be a `const CommandParameters&` then? or at least, this is the dual of `SaveSettings`, and the ref to Settings is const there

Reply:
You are right, but when I try to do that, there is some strange code in Nyquist.cpp that is mutating the parameters sometimes.  Probably unnecessarily, and I will figure out a fix.  I have issue #2578 to address these const correctness issues later.